bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/handler/VehicleHandler.java

```diff
@@ -12,167 +12,240 @@
  *
+ * the introduction of channelToBeUpdated is ugly, but if there is a refresh of one channel, always all channels were
+ * updated
+ *
  * @author Bernd Weymann - Initial contribution
  * @author Norbert Truchsess - edit and send charge profile
+ * @author Martin Grassl - refactoring, merge with VehicleChannelHandler
+ * @author Mark Herwege - refactoring, V2 API charging
  */
 @NonNullByDefault
-public class VehicleHandler extends VehicleChannelHandler {
+public class VehicleHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(VehicleHandler.class);
+
+    private boolean hasFuel = false;
+    private boolean isElectric = false;
+    private boolean isHybrid = false;
+
+    // List Interfaces
+    private volatile List<RequiredService> serviceList = List.of();
+    private volatile String selectedService = Constants.UNDEF;
+    private volatile List<CheckControlMessage> checkControlList = List.of();
+    private volatile String selectedCC = Constants.UNDEF;
+    private volatile List<ChargingSession> sessionList = List.of();
+    private volatile String selectedSession = Constants.UNDEF;
+
+    private MyBMWCommandOptionProvider commandOptionProvider;
+    private LocationProvider locationProvider;
+    private ZoneId timeZone;
+
+    // Data Caches
+    private Optional<VehicleStateContainer> vehicleStatusCache = Optional.empty();
+    private Optional<byte[]> imageCache = Optional.empty();
+
     private Optional<MyBMWProxy> proxy = Optional.empty();
-    private Optional<RemoteServiceHandler> remote = Optional.empty();
-    public Optional<VehicleConfiguration> configuration = Optional.empty();
+    private Optional<RemoteServiceExecutor> remote = Optional.empty();
+    private Optional<MyBMWVehicleConfiguration> vehicleConfiguration = Optional.empty();
     private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
     private Optional<ScheduledFuture<?>> editTimeout = Optional.empty();
 
     private ImageProperties imageProperties = new ImageProperties();
-    VehicleStatusCallback vehicleStatusCallback = new VehicleStatusCallback();
-    ChargeStatisticsCallback chargeStatisticsCallback = new ChargeStatisticsCallback();
-    ChargeSessionsCallback chargeSessionCallback = new ChargeSessionsCallback();
-    ByteResponseCallback imageCallback = new ImageCallback();
 
-    public VehicleHandler(Thing thing, MyBMWCommandOptionProvider cop, LocationProvider lp, String driveTrain) {
-        super(thing, cop, lp, driveTrain);
-    }
+    public VehicleHandler(Thing thing, MyBMWCommandOptionProvider commandOptionProvider,
+            LocationProvider locationProvider, TimeZoneProvider timeZoneProvider, String driveTrain) {
+        super(thing);
+        logger.trace("xxxVehicleHandler.constructor {}, {}", thing.getUID(), driveTrain);
+        this.commandOptionProvider = commandOptionProvider;
+        this.timeZone = timeZoneProvider.getTimeZone();
```

Comment:
It would be better to store a reference to `TimeZoneProvider` instead and call `getTimeZone()` each time you need the time-zone. Then current time-zone will always be taken into consideration, also after a configuration change or change in DST.

Reply:
good point. I was unsure which way to go and it looks like I chose the wrong one - fixed.